### ISSUE TITLE
Console errors about `oneOfType`

### ISSUE THREAD

**Issue by patik** [reactjs/react-modal#420]
### Summary:

Using v2.0.1, I get this error in the console, printed four times in a row:

```
Warning: Invalid argument supplid to oneOfType. Expected an array of check functions, but received undefined at index 0.
```

It seems to trace back to `function createUnionTypeChecker()`, specifically the `if (typeof checker !== 'function')` conditional.  When I log `checker` at that point it's undefined. Here's what `arrayOfTypeCheckers` looks like:

![image](https://user-images.githubusercontent.com/262137/27232269-6d074a78-5283-11e7-92c5-d7fb67ed197d.png)

### Steps to reproduce:

 1. Use react-modal v2.0.1 and React v15.6.1
 1. Implement the [minimal example from the docs](https://reactcommunity.org/react-modal/examples/minimal.html)
 1. Notice that the parent component doesn't render because ReactModal returned an object
     + Console error from React v15.6.1: `Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: object.`
 
### Expected behavior:

No console errors, and the component renders

### Additional notes:

![image](https://user-images.githubusercontent.com/262137/27232104-cd7d2e82-5282-11e7-8413-259987298c35.png)

**Comment by diasbruno** [comment]
Hi, PR #419 should fix it. Thanks.

**Comment by diasbruno** [comment]
Released `v2.0.2`.
